### PR TITLE
Create DictationHandler script for easy access to dictation service

### PR DESCRIPTION
**Pull request by luis-valverde-ms**
Created a script to provide basic access to the dictation service without having to resort to scripting, like SpeechInputHandler does for the speech service.

### CODE REVIEW

**Comment by wiwei** [comment]
Note that this should probably be going into mrtk_development instead of mrtk_development_hl2, now that the HL2 payload is in mrtk_development.

File: Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/DictationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        /// Event raised when the recognizer stops. Contains the final recognized string.
+        /// </summary>
+        public StringUnityEvent OnDictationComplete;
+
+        /// <summary>
+        /// Event raised when an error occurs. Contains the string representation of the error reason.
+        /// </summary>
+        public StringUnityEvent OnDictationError;
+
+        private bool isRecording = false;
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/DictationHandler.cs]: > private bool isRecording = false; [](start = 7, length = 34)

Is this intended to be used for logging/debugging purposes only? It looks like this is getting set correctly but not inspected.

Comment (wiwei) [reply Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/DictationHandler.cs]: Or was this kinda to make sure that StartRecording doesn't invoke dictationSystem.StartRecording in the case it gets double-called?

---
In reply to: [271491908](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3764#discussion_r271491908) [](ancestors = 271491908)

Comment (luis-valverde-ms) [reply Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/DictationHandler.cs]: The second. It must have been lost when refactoring. Thanks for the catch!

**Pull request merged**